### PR TITLE
Adds support for JSON5 files

### PR DESCRIPTION
**Pull request by giacomorebonato**
Basic implementation for supporting JSON5 file format.
Monaco can be probably better fine tuned for the purpose, but this is the best I can do at the moment with my knowledge.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,6 +71,7 @@
     "eslint-config-prettier": "^8.10.0",
     "eslint-plugin-prettier": "^5.0.0",
     "eslint-plugin-unused-imports": "^3.0.0",
+    "monaco-editor": "^0.43.0",
```
Comment (AykutSarac) [inline package.json:74]: Do we use this package anywhere?

Comment (AykutSarac) [reply package.json:74]: Also we don't need package-lock.json, we use yarn.

File: src/components/MonacoEditor/index.tsx [hunk 2]
```diff
@@ -46,11 +58,16 @@ export const MonacoEditor = () => {
           {
             fileMatch: ["*"],
             schema: jsonSchema,
+            uri: "",
           },
         ],
       }),
     });
```
Comment (AykutSarac) [inline src/components/MonacoEditor/index.tsx:65]: What is the purpose of this change here?

File: src/components/MonacoEditor/index.tsx [hunk 3]
```diff
@@ -35,9 +27,29 @@ export const MonacoEditor = () => {
...
+      checkJs: false,
+      allowUnusedLabels: true,
+    });
+    monaco?.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
+      noSemanticValidation: true,
+      noSyntaxValidation: true,
+    });
+
     monaco?.languages.json.jsonDefaults.setDiagnosticsOptions({
+      ...monaco?.languages.json.jsonDefaults.diagnosticsOptions,
```
Comment (AykutSarac) [inline src/components/MonacoEditor/index.tsx:52]: What is the purpose of this change here?

File: package.json (hunk 1 above)
Comment (giacomorebonato) [reply package.json:74]: I added `monaco-editor` because it helped me with intellisense and types. I am taking it out to keep this PR more scoped

File: src/components/MonacoEditor/index.tsx (hunk 2 above)
Comment (giacomorebonato) [reply src/components/MonacoEditor/index.tsx:65]: This was to solve a type issue after I referenced `monaco-editor` types. Removing it to keep this PR more scoped. 

Screenshot 2023-09-17 at 23 10 58

File: src/components/MonacoEditor/index.tsx (hunk 3 above)
Comment (giacomorebonato) [reply src/components/MonacoEditor/index.tsx:52]: I am adding a comment in the code for this. JSON5 has valid JS syntax, but because the files don't have any exports, it triggers some warnings in Monaco UI if this setting is not there

File: package.json (hunk 1 above)
Comment (giacomorebonato) [reply package.json:74]: package-lock added by mistake. Removing it
